src/platform_impl/windows/window.rs

```diff
@@ -881,6 +882,32 @@ impl Window {
     unsafe { set_skip_taskbar(self.hwnd(), skip) };
   }
 
+  #[inline]
+  pub fn set_taskbar_progress_state(&self, state: i32) {
+    let handle = self.window.0;
+
+    unsafe {
+      let taskbar_list: ITaskbarList = CoCreateInstance(&TaskbarList, None, CLSCTX_SERVER).unwrap();
+
+      taskbar_list
+        .SetProgressState(handle, TBPFLAG(state))
+        .unwrap_or(());
+    }
+  }
+
+  #[inline]
+  pub fn set_taskbar_progress(&self, current: f64) {
+    let handle = self.window.0;
+
+    unsafe {
+      let taskbar_list: ITaskbarList = CoCreateInstance(&TaskbarList, None, CLSCTX_SERVER).unwrap();
+
+      taskbar_list
+        .SetProgressValue(handle, current as u64, 100)
+        .unwrap_or(());
+    }
+  }
```

Comment:
combine theses into one function `set_progress_bar` that takes `ProgressBarState` and conditionally check for the `Some` fields and set it